chia/wallet/wallet_state_manager.py

```diff
@@ -1116,7 +1116,7 @@ async def add_coin_states(
                     # if the coin has been spent
                     elif coin_state.created_height is not None and coin_state.spent_height is not None:
                         self.log.debug("Coin Removed: %s", coin_state)
-                        children: Optional[List[CoinState]] = None
```

Comment:
this is a very long function (a separate issue). It's not trivial to tell that the patch has identical behavior. It looks like `fork_height`,  `peer` and `coin_name` is not mutated, which is a good start. Now, is `children` used unconditionally in this scope?

It's used if `record is None` and it looks like the other call is at the same indentation level as the new call. So this might save us a call. Unless there are any early exits. It looks like the only possibility of an early exit would be an exception.

Reply:
> this is a very long function (a separate issue)

Yes im working towards fixing, this PR here is also kind of preparation for this and https://github.com/Chia-Network/chia-blockchain/pull/14870 is also meant to be a step in that direction. 

> Now, is `children` used unconditionally in this scope?

From what i see when searching for it in this function, yes.

> It looks like the only possibility of an early exit would be an exception.

Yeah and for this case it probably doesn't make sense to delay it.